tests/test_api_urls.py

```diff
@@ -42,11 +48,17 @@ def test_api_issues_search(self):
     def test_api_issues_out_of_range(self):
         '''API issue for a non existent number returns JSON 404.'''
         # If we reach 1,000,000 webcompat issues we can celebrate
-        rv = self.app.get('/api/issues/1000000', environ_base=headers)
-        json_body = json.loads(rv.data)
-        self.assertEqual(rv.status_code, 404)
-        self.assertEqual(rv.content_type, 'application/json')
-        self.assertEqual(json_body['status'], 404)
+        with patch('requests.get') as github_data:
+            api_response = MagicMock(spec=Response)
+            api_response.status_code = 404
+            api_response.content_type = 'application/json'
+            api_response.data = '{"message":"Not Found. Lost in Punk Cat Space","status":404}'  # nopep8
```

Comment:
I think I understand your [general comment](https://github.com/webcompat/webcompat.com/pull/1697) about testing github API. And this line helped me to clear it up. Here the mock is trying to replicate what webcompat.com sends… but it's not what GitHub is sending. Basically you are testing a fabricated answer of what you are expecting. So the test will always be valid… and indeed useless.

That said, GitHub is sending something totally different. Let me show you:

if I do, aka a direct call to the GitHub API

```bash
http GET https://api.github.com/repos/webcompat/webcompat-tests/issues/100000
```

GitHub sends:

```http
HTTP/1.1 404 Not Found
Access-Control-Allow-Origin: *
Access-Control-Expose-Headers: ETag, Link, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval
Content-Encoding: gzip
Content-Security-Policy: default-src 'none'
Content-Type: application/json; charset=utf-8
Date: Thu, 03 Aug 2017 00:14:08 GMT
Server: GitHub.com
Status: 404 Not Found
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
Transfer-Encoding: chunked
X-Content-Type-Options: nosniff
X-Frame-Options: deny
X-GitHub-Media-Type: github.v3; format=json
X-GitHub-Request-Id: F745:2AD24:33D497E:764441A:59826ACF
X-RateLimit-Limit: 60
X-RateLimit-Remaining: 59
X-RateLimit-Reset: 1501722848
X-Runtime-rack: 0.029843
X-XSS-Protection: 1; mode=block

{
    "documentation_url": "https://developer.github.com/v3", 
    "message": "Not Found"
}
```

so your mock, should be 

```python
api_response.data = '{"documentation_url": "https://developer.github.com/v3", "message": "Not Found"}'
```

and we should be testing that we receive the right json_body aka `{"message":"Not Found. Lost in Punk Cat Space","status":404}`

Let's test this idea.

```diff
diff --git a/tests/test_api_urls.py b/tests/test_api_urls.py
index 2751c22..9e8e179 100644
--- a/tests/test_api_urls.py
+++ b/tests/test_api_urls.py
@@ -52,13 +48,15 @@ class TestAPIURLs(unittest.TestCase):
             api_response = MagicMock(spec=Response)
             api_response.status_code = 404
             api_response.content_type = 'application/json'
-            api_response.data = '{"message":"Not Found. Lost in Punk Cat Space","status":404}'  # nopep8
+            api_response.data = '{"documentation_url": "https://developer.github.com/v3", "message": "Not Found"}'  # nopep8
             github_data.return_value = api_response
-            rv = self.app.get('/api/issues/1000000', environ_base=headers)
+            rv = self.app.get('/api/issues/1', environ_base=headers)
             json_body = json.loads(rv.data)
             self.assertEqual(rv.status_code, 404)
             self.assertEqual(rv.content_type, 'application/json')
             self.assertEqual(json_body['status'], 404)
+            self.assertEqual(json_body['message'],
+                             'Not Found. Lost in Punk Cat Space')
 
     def test_api_wrong_route(self):
         '''API with wrong route returns JSON 404.'''
```

and let's run the test again.

```
→ nosetests tests/test_api_urls.py
.........
----------------------------------------------------------------------
Ran 9 tests in 0.019s

OK
```

You see the tests are passing. We are really testing the webcompat API. 

So what you need to do is to fix the mockup everywhere to replicate what **GitHub is sending** not what webcompat is sending. :) and then tests what Webcompat is supposed to answer for each of these cases.

I'll let you do that. Before doing another round of review.